designer/server/src/routes/forms/create.js

```diff
@@ -198,7 +200,13 @@ export default [
       // Submit new form metadata
       try {
         if (!result.error) {
-          await forms.create(result.value)
+          await forms.create({
+            metadata: result.value,
+            author: {
+              id: user.id,
+              displayName: user.displayName
+            }
+          })
```

Comment:
Once we've tightened the check for `credentials.user` we can destructure it without worrying that it might be undefined sometimes

This is using the 2x param suggestion for `forms.create(metadata, author)`

```suggestion
          const { id, displayName } = credentials.user
          await forms.create(result.value, { id, displayName })
```

Reply:
> Once we've tightened the check for credentials.user

We need to come back to look at this.
For now, I've added an exception with @ts-expect-error for the user is not undefined case.